src/libraries/System.Text.Json/src/System/Text/Json/ThrowHelper.Serialization.cs

```diff
@@ -471,6 +471,13 @@ public static void ThrowJsonException_MetadataReferenceObjectCannotContainOtherP
             ThrowJsonException_MetadataReferenceObjectCannotContainOtherProperties();
         }
 
+        [DoesNotReturn]
+        public static void ThrowJsonException_MetadataUnexpectedProperty(ReadOnlySpan<byte> propertyName, ref ReadStack state)
+        {
+            state.Current.JsonPropertyName = propertyName.ToArray();
+            ThrowJsonException(SR.Format(SR.MetadataUnexpectedProperty));
```

Comment:
why are you assigning property name if you're not using it? Should you be passing it to the exception? Btw Should you be converting UTF-8 bytes to string?

Reply:
That field is picked up by the exception handling infrastructure when building the `JsonPath` for the exception. It's not a pattern that this change is introducing.